src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -730,7 +756,7 @@ def is_read_only(self) -> bool:
     def recording(self) -> BaseRecording:
         if not self.has_recording():
             raise ValueError("SortingAnalyzer could not load the recording")
-        return self._recording
+        return self._temporary_recording or self._recording
```

Comment:
We should use  has_temporary_recording here also.
no ?

Reply:
done